Lib/test/test_tools/test_pathfix.py

```diff
@@ -0,0 +1,48 @@
+import unittest, os, subprocess
+from test.test_tools import import_tool, scriptsdir
+import sys
+from test import support
+
+
+class TestPathfixFunctional(unittest.TestCase):
+    script = os.path.join(scriptsdir, 'pathfix.py')
+
+    def setUp(self):
+        self.temp_file = support.TESTFN
+        self.addCleanup(support.unlink, support.TESTFN)
+
+    def pathfix(self, shebang, pathfix_flags):
+        with open(self.temp_file, 'w') as f:
+            f.write(f'{shebang}\n' + 'print("Hello world")\n')
+        subprocess.call([sys.executable, self.script, *pathfix_flags, self.temp_file])
+        with open(self.temp_file) as f:
+            output = f.read()
+        print(output)
+        return output
+
+    def test_pathfix_keeping_flags(self):
+        self.assertEqual(
+            self.pathfix(
+                '#! /usr/bin/env python -R',
```

Comment:
nitpick: instead of "/usr/bin/env python" and "/usr/bin/python" ... which is Python 2 in most Linux distributions... I would prefer to use "/usr/bin/env python" and "/usr/bin/python3" which is Python 3 in all Linux distributions :-)